src/pages/home/report/ReportView.js

```diff
@@ -4,40 +4,51 @@ import PropTypes from 'prop-types';
 import ReportActionsView from './ReportActionsView';
 import ReportActionCompose from './ReportActionCompose';
 import {addAction} from '../../../libs/actions/Report';
+import compose from '../../../libs/compose';
 import KeyboardSpacer from '../../../components/KeyboardSpacer';
 import styles from '../../../styles/styles';
 import SwipeableView from '../../../components/SwipeableView';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoading';
+import withDrawerState from '../../../components/withDrawerState';
+import withWindowDimensions from '../../../components/withWindowDimensions';
 
 const propTypes = {
-    // The ID of the report actions will be created for
+    /* The ID of the report the selected report */
     reportID: PropTypes.number.isRequired,
+
+    /* Is the view ready to be displayed */
+    isReady: PropTypes.bool.isRequired,
+
+    /* Is the report view covered by the drawer */
+    isDrawerOpen: PropTypes.bool.isRequired,
+
+    /* Is the window width narrow, like on a mobile device */
+    isSmallScreenWidth: PropTypes.bool.isRequired,
 };
 
-// This is a PureComponent so that it only re-renders when the reportID changes or when the report changes from
-// active to inactive (or vice versa). This should greatly reduce how often comments are re-rendered.
-class ReportView extends React.Component {
-    shouldComponentUpdate(prevProps) {
-        return this.props.reportID !== prevProps.reportID;
-    }
-
-    render() {
-        return (
-            <View style={[styles.chatContent]}>
-                <ReportActionsView
-                    reportID={this.props.reportID}
+function ReportView(props) {
+    const isComposeDisabled = props.isDrawerOpen && props.isSmallScreenWidth;
+
+    return (
+        <View key={props.reportID} style={[styles.chatContent]}>
+            {!props.isReady && <FullScreenLoadingIndicator />}
+
+            <ReportActionsView reportID={props.reportID} />
+
+            <SwipeableView onSwipeDown={() => Keyboard.dismiss()}>
+                <ReportActionCompose
+                    isDisabled={isComposeDisabled}
+                    onSubmit={text => addAction(props.reportID, text)}
+                    reportID={props.reportID}
                 />
-                <SwipeableView onSwipeDown={() => Keyboard.dismiss()}>
-                    <ReportActionCompose
-                        onSubmit={text => addAction(this.props.reportID, text)}
-                        reportID={this.props.reportID}
-                        key={this.props.reportID}
-                    />
-                </SwipeableView>
-                <KeyboardSpacer />
-            </View>
-        );
-    }
+            </SwipeableView>
```

Comment:
Hiding the Keyboard by unmounting this field turned out to be a big problem on Android: https://github.com/Expensify/Expensify.cash/pull/2221#issuecomment-815093078

"Disabling" the field works on all platforms but Android. The keyboard would still stay open 

Other suggested fixes don't work as well (for Android) 
 - calling `Keyboard.dismiss()` when we go back to LHN, briefly closes the keyboard only to bring it back up when you are at LHN

You'll probably want me to either remove this "fix" altogether or keep it inside ReportActionCompose and don't expose it outside

I've exhausted all my options to deal with this in the current PR. This looks like to be something deeper and needs more investigation. I would ask if it can be handled outside of the current branch in a separate issue. Other than that the work on the transitions is complete

Reply:
Ok, thanks for all the details. I'll test around a bit while I review. I'm not sure I totally get the problem (but might just need to read your explanation a few more times to get it).

Reply:
If you're testing you can checkout from the previous commit: https://github.com/Expensify/Expensify.cash/pull/2221/commits/7847ffa0b1330f5656ebdb91859eb2cf51412f4e
The code in `ReportView` is setup to unmount the compose input which is how I would get the error when I tap back in the header to go to the LHN

Reply:
Ah ok I see. Yes, let's just not worry about this for now. As long as everything in this PR is working well I'm happy 🙃